gke/quickstart/multitenant/main.tf

```diff
@@ -0,0 +1,232 @@
+  location              = google_container_cluster.default.fleet[0].membership_location
+}
+# [END gke_quickstart_multitenant_cluster]
+# [START gke_quickstart_multitenant_rbac]
+resource "google_gke_hub_scope_rbac_role_binding" "default" {
+  for_each = local.teams
+
+  project                    = data.google_project.default.project_id
+  scope_rbac_role_binding_id = each.key
+  scope_id                   = google_gke_hub_scope.default[each.key].scope_id
+  user                       = google_service_account.default[each.key].email
+  role {
+    predefined_role = "EDIT"
+  }
+  depends_on = [google_gke_hub_scope.default]
+}
+# [END gke_quickstart_multitenant_rbac]
+# [START gke_quickstart_multitenant_database]
+// Create Database for Apps
+resource "google_sql_database" "database" {
+  name     = "multitenant-app"
+  project  = data.google_project.default.project_id
+  instance = google_sql_database_instance.default.name
+}
+
+resource "google_sql_database_instance" "default" {
+  name             = "gkee-multitenant-app-db"
+  project          = data.google_project.default.project_id
+  region           = "us-central1"
+  database_version = "MYSQL_8_0"
+  settings {
+    tier = "db-f1-micro"
+  }
+  # set `deletion_protection` to true, will ensure that one cannot accidentally delete this instance by
+  # use of Terraform whereas `deletion_protection_enabled` flag protects this instance at the GCP level.
+  deletion_protection = false
+}
+
+resource "google_sql_user" "users" {
+  name     = "multitenant-app"
+  project  = data.google_project.default.project_id
+  instance = google_sql_database_instance.default.name
+  host     = "cloudsqlproxy~%"
+  password = ""
+}
+
+resource "google_project_iam_member" "cloudsql" {
+  for_each = local.teams
+
+  project = data.google_project.default.project_id
+  role    = "roles/cloudsql.client"
+  member  = "serviceAccount:${data.google_project.default.project_id}.svc.id.goog[${each.value}-team/default]"
+}
+
+data "google_client_config" "default" {}
+
+provider "kubernetes" {
+  host                   = "https://${google_container_cluster.default.endpoint}"
+  token                  = data.google_client_config.default.access_token
+  cluster_ca_certificate = base64decode(google_container_cluster.default.master_auth[0].cluster_ca_certificate)
+}
+
+// Store Database instance name in team config maps
+resource "kubernetes_config_map" "default" {
+  for_each = local.teams
+
+  metadata {
+    name      = "database-configmap"
+    namespace = "${each.key}-team"
+  }
+
+  data = {
+    CONNECTION_NAME = google_sql_database_instance.default.connection_name
+  }
+}
+# [END gke_quickstart_multitenant_database]
+# [END gke_quickstart_multitenant]
+
+// Included for automated testing
+resource "time_sleep" "wait_5_minutes" {
+  depends_on = [
+    google_container_cluster.default,
+    google_gke_hub_namespace.default
+  ]
+
+  create_duration = "5m"
+}
+
+module "gcloud" {
```

Comment:
Per samples style we cannot use modules. IIUC this is for a sample app - could we prebuilt and publish the container (or use an existing one) and use `kubernetes_deployment` instead?

https://github.com/terraform-google-modules/terraform-docs-samples/blob/bf504eeef1b30e42f2df63f35e0052bb740825dc/CONTRIBUTING.md?plain=1#L39-L45

Reply:
Hi @bharathkkb - This is using an existing module (not defining a new module), which I do believe is permitted? https://github.com/terraform-google-modules/terraform-docs-samples/blob/bf504eeef1b30e42f2df63f35e0052bb740825dc/CONTRIBUTING.md?plain=1#L42

Also this section is outside the snippet for build coverage.  In the actual quickstart the user will impersonate a team's service account to kubectl apply the relevant manifest.  So use of this module use isn't included in the user's journey.

I was looking at existing sample app images, but so far hadn't identified a good candidate for this particular use case.  That said if we identify one we can switch it anytime, or long term switch to a pre-built a container.  But we'd still want the user to apply the manifest's as different teams/service accounts.

Reply:
To expedite, we could remove the  `module "gcloud"`.  Other than successfully applying the manifest, this doesn't include actual testing of the apps, so honestly the value is somewhat limited.

Reply:
@apeabody I believe for samples in this repo we don't want to have modules at all. An alternative approach is to have them as an example within one of our modules with region tags like https://github.com/terraform-google-modules/terraform-google-network/blob/master/examples/simple_project/main.tf#L23 

Yup removing having comment SGTM!

Reply:
SG @bharathkkb, removed the module.